boto3/resources/factory.py

```diff
@@ -246,16 +246,28 @@ def _create_reference(factory_self, name, reference, service_name,
         # References are essentially an action with no request
         # or response, so we can re-use the response handlers to
         # build up resources from identifiers and data members.
-        handler = ResourceHandler('', factory_self, resource_defs,
-                                  service_model, reference.resource)
+        handler = ResourceHandler(reference.resource.path, factory_self,
+                                  resource_defs, service_model,
+                                  reference.resource)
+
+        # Are there any identifiers that need access to data members?
+        # This is important when building the resource below since
+        # it requires the data to be loaded.
+        needs_data = False
+        if [i for i in reference.resource.identifiers if i.source == 'data']:
+            needs_data = True
```

Comment:
Isn't this just `needs_data = any(i.source == 'data' for i in reference.resource.identifiers)`